### PR TITLE
Fix KeyError when removing object that is not referenced in ordering annotation

### PR DESCRIPTION
**Pull request by vangheem**

### CODE REVIEW

**Comment by vangheem** [comment]
@plone/framework-team any idea why jenkins isn't working?

**Comment by ale-rt** [comment]
@vangheem I think this was a temporary glitch, I started a PR just now https://jenkins.plone.org/job/pull-request-4.3/256/

File: src/plone/folder/default.py [hunk 1]
```diff
@@ -34,8 +34,11 @@ def notifyRemoved(self, obj_id):
         """ see interfaces.py """
         order = self._order()
         pos = self._pos()
-        idx = pos[obj_id]
-        del order[idx]
+        try:
+            idx = pos[obj_id]
+            del order[idx]
+        except KeyError:
+            pass
```
Comment (ale-rt) [inline src/plone/folder/default.py]: I do not like idx stays undefined if the exception is raised. 
`del order[pos[obj_id]]` or  `order.pop(pos[obj_id], None)` would avoid the need of the temp idx variable.
Anyway not a blocker.

**Comment by ale-rt** [comment]
@vangheem it would be great if this could be ported to master!

**Comment by vangheem** [comment]
Sweet, thanks @ale-rt!

I'll make another PR for master.

**Comment by ale-rt** [comment]
Thanks to you!

**Pull request merged**